packages/core/src/actions/cell/insert.ts

```diff
@@ -129,6 +129,14 @@ export const insertCellRightInline = insert(CELL_INSERT_INLINE_RIGHT);
 const newIds = ({ id, ...item }: Partial<Cell>) => {
   return {
     ...item,
+    content: item.content && {
+      plugin: item.content.plugin,
+      state: JSON.parse(JSON.stringify(item.content.state)),
+    },
+    layout: item.layout && {
+      plugin: item.layout.plugin,
+      state: JSON.parse(JSON.stringify(item.layout.state)),
+    },
```

Comment:
This is what fixes the duplicate button problem mentioned in #802 . Sorry again about mixing PRs.

Reply:
no worries :-) looks good. This would probably fail if some state happens to be circular (which should not be the case in my opinion)

Reply:
Agreed! Cheers